src/sagemaker/hyperpod/training/hyperpod_pytorch_job.py

```diff
@@ -168,7 +178,14 @@ def _process_replica_resources(cls, data):
                 elif NEURON_RESOURCE_KEY in requests_values:
                     acc_lim = requests_values[NEURON_RESOURCE_KEY]
 
-            limits_values = _get_limits(instance_type, vcpu_limit, memory_limit, acc_lim, accelerator_partition_type, acc_partition_lim)
+                if efa is not None:
+                    requests_values["vpc.amazonaws.com/efa"] = efa
```

Comment:
1/ If there is no values provided, we use values from the instance. But looks like here you set requests_values. It is not a symmetric operation. Can you check ident in this block
2/ Do you want to add EFA_RESOURCE_KEY constant?